src/error.ts

```diff
@@ -83,14 +87,14 @@ export function createError(
   }
 
   if (input.statusCode) {
-    err.statusCode = input.statusCode;
+    err.statusCode = sanetizeStatusCode(input.statusCode, err.statusCode);
   } else if (input.status) {
-    err.statusCode = input.status;
+    err.statusCode = sanetizeStatusCode(input.status, err.statusCode);
   }
   if (input.statusMessage) {
-    err.statusMessage = input.statusMessage;
+    err.statusMessage = sanetizeStatusMessage(input.statusMessage);
```

Comment:
I didn't think we needed to sanitize the error object itself, only when it was set to the response object.

Reply:
Inddeed. Doing this as extra caution in case someone is picking statusMessage for custom setter.

Reply:
I think we should leave it up to them. Maybe their error message is in Arabic (see example here: https://github.com/nuxt/nuxt/issues/14688). They might want to be able to read that error message elsewhere, or display it (as, for example in Nuxt we would do in the page).

Reply:
I think this is the purpose of `message` right? `status{code,message}` are purposed for HTTP headers.

Reply:
I think it still makes sense to leave it up to them. If they are using a custom setter, then they might want to sanitise differently - e.g. replace with `-`.

Reply:
If the input is already santized (and includes valid chars) we wouldn't remove them. 

In general `statusMessage` should probably be used less IMO. HTTP clients (browser, node, curl, etc) already have defaults matching the code.

For anything else (display purpose), `message` is there which we also need to sanetize from xss later.

Reply:
I get that probably people _should_ do that. But this is very much an edge case, as we will always sanitise it _ourselves_ in h3, nitro or nuxt before setting the response status.

The only remaining situation is where the _user_ sets the header based on `error.statusMessage`. And for that purpose we export utilities they can use to sanitise the header. Sanitising it at the creation of the error itself is a breaking change that may affect people more than we expect, including breaking error pages in Nuxt if people were using `statusMessage` rather than `message`.

Reply:
Looking at linked nuxt issue, throw error with unsupported characters was always broken (please correct me if i am wrong ) 

message is standard error property also it is nothing new.

i mainly insist on doing this to avoid breaking changes from now on that we are normalizing errorMessage field, it should be consistent and universally :) Also that thinking to support a standard response object to h3 (similar to ipx generic response object) for that we also have similar concern and if not fixing a standard normalization now, it would be lots of inconsistency on a field that only in one case is not normalized.

Reply:
I'm not only thinking about throwing the error on server side, but also on client. Or throwing an error only to have it bubble up and be handled elsewhere in the app.

Reply:
Hmm i see it is only for showerror. Well we need to fix it anyway. Status message was never meant for that evne if wrongly used.

I think we can make it a warning only for nuxt wdyt? (When normalized and input status message are not the same warn to use message field instead )

Reply:
It just feels like extra work to me - we are normalising statusMessage twice. Once when we create error and once when we set statusMessage. And the error doesn't _need_ normalising separately if it is being used for the envisioned purpose. The only reason it needs to be normalised if it is _not_ being used for the normal purpose, such as if the user is going to directly set the response statusMessage from it.

Reply:
The status* fields on error object specify exist because of http response so it is not wrong to normalize them.

it comes to consistency and external usage of error object. 

If it is not normalized today, breaking change surface of normalizing this field will extend in the future where we cannot easily do this anymore. Today only issue is with (wrong) usages only and we dan soft warn for a while and ensure normalization in long term. I agree with your concern of util currently being used in places out of http response and warning seems a better way to move usage to the correct message field.

Reply:
Updated with warn.

Reply:
Also made soft warn for now.